From 4026b23fc8e0ff2e50e9ee7d6249c8d622ee0aba Mon Sep 17 00:00:00 2001
From: Peter Kasting <pkasting@chromium.org>
Date: Wed, 28 Sep 2022 01:57:53 +0000
Subject: [PATCH] Enable C++20 on Linux.

Doing this in advance of enabling for more platforms allows monitoring
compile-time effects with less chance of adversely affecting the CQ,
while still getting the benefit of breaking a bot if someone lands a
change that breaks the build only in C++20.

Bug: 1284275
Change-Id: I88d6db79fad8d4552a2ee68b5d5b54914ca61042
Cq-Include-Trybots: chromium/try:linux_chromium_cfi_rel_ng
Cq-Include-Trybots: chromium/try:linux_chromium_msan_rel_ng
Cq-Include-Trybots: chromium/try:linux-official
Cq-Include-Trybots: chromium/try:linux-swangle-try-x64
Cq-Include-Trybots: chrome/try:linux-chrome
Cq-Include-Trybots: chrome/try:linux-pgo
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3880023
Reviewed-by: Andrew Grieve <agrieve@chromium.org>
Commit-Queue: Peter Kasting <pkasting@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1052163}
---

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 249cf0b..dcf4d10 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -592,6 +592,9 @@
       if (is_clang) {
         cflags_cc += [ "-fno-trigraphs" ]
       }
+    } else if (is_linux) {
+      # TODO(crbug.com/1284275): Switch to C++20 on all platforms.
+      cflags_cc += [ "-std=${standard_prefix}++20" ]
     } else {
       cflags_cc += [ "-std=${standard_prefix}++17" ]
     }
